### PR TITLE
Add support for checking and installing Jenkins plugins

### PR DESCRIPTION
**Pull request by PhilipGough**
ping @wei-lee @laurafitzgerald

### CODE REVIEW

File: configure-buildfarm/defaults/main.yml [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
 # Valid platforms are 'android' and 'ios'. Multiple values should be separated by comma
 project_name: jenkins
 buildfarm_version: "2.0"
+
+jenkins_home: /var/lib/jenkins
+jenkins_plugins:
+  -
+    name: "ssh-slaves"
+    version: 1.20
+    archive: "http://updates.jenkins-ci.org/download/plugins/ssh-slaves/1.20/ssh-slaves.hpi"
```
Comment (wei-lee) [inline configure-buildfarm/defaults/main.yml]: can those values be overridden in the rhmap-ansible repo? We might have to use productised version of them there.

Comment (PhilipGough) [inline configure-buildfarm/defaults/main.yml]: Yes, these values are coming from role defaults file which takes least precedence of all variables and so can be overridden anywhere

**Comment by PhilipGough** [comment]
@wei-lee This would be best verified by deleting some or all of the plugins in the list and running the installer, you can use `--tags=plugins` to hit only the required tasks if you prefer. A second run of the same command should see nothing installed

File: configure-buildfarm/defaults/main.yml [hunk 2]
```diff
@@ -1,6 +1,23 @@
 ---
 
-# The platfroms that are supported by the self-managed build farm. 
+# The platfroms that are supported by the self-managed build farm.
```
Comment (laurafitzgerald) [inline configure-buildfarm/defaults/main.yml]: platforms*

**Comment by wei-lee** [comment]
@PhilipGough I have added a new commit to add the missing "-remoting" flag. See https://github.com/aerogear/digger-installer/pull/19/commits/d6caf6a7b44d5adea6c4ea944ac3035e57511d68.

Also when the Jenkins server is being restarted, I get an error like this:

```
WARNING: Unexpected exception occurred while performing restart command.
java.lang.NullPointerException
	at jenkins.model.Jenkins.doRestart(Jenkins.java:4116)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.cli.declarative.MethodBinder.call(MethodBinder.java:114)
	at hudson.cli.declarative.CLIRegisterer$1.main(CLIRegisterer.java:224)
	at hudson.cli.CliManagerImpl.main(CliManagerImpl.java:95)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.perform(RemoteInvocationHandler.java:895)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:870)
	at hudson.remoting.RemoteInvocationHandler$RPCRequest.call(RemoteInvocationHandler.java:829)
	at hudson.remoting.UserRequest.perform(UserRequest.java:153)
	at hudson.remoting.UserRequest.perform(UserRequest.java:50)
	at hudson.remoting.Request$2.run(Request.java:336)
	at hudson.remoting.InterceptingExecutorService$1.call(InterceptingExecutorService.java:68)
	at hudson.cli.CliManagerImpl$1.call(CliManagerImpl.java:66)
	at hudson.remoting.CallableDecoratorAdapter.call(CallableDecoratorAdapter.java:18)
	at hudson.remoting.CallableDecoratorList$1.call(CallableDecoratorList.java:21)
	at jenkins.util.ContextResettingExecutorService$2.call(ContextResettingExecutorService.java:46)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
```

Looks like it is just an warning and the Jenkins server is actually restarted. 

I think it might be better if we ignore the error some how, or just print out the warning but let the ansible command continue?

File: configure-buildfarm/tasks/main.yml [hunk 3]
```diff
@@ -53,20 +76,29 @@
...
+
 -
   name: "Get image version"
   set_fact:
     android_slave_image: "{{ android_slave_image_def[0][0].image }}"
+  tags:
+     - config
 
 -
   name: "Create podtemplate update Groovy script from template"
```
Comment (laurafitzgerald) [inline configure-buildfarm/tasks/main.yml]: Is there a `config` tag required for this task?

**Comment by PhilipGough** [comment]
@wei-lee looking at solution for above error

**Pull request merged**